### ISSUE TITLE
Implement "Level" result for LUA access rules

### ISSUE THREAD

**Issue by black-sliver** [black-sliver/PopTracker#11]
* LUA access rules (`$function`) should be able to return the level of accessibility as second result.
* Define constants in LUA global `AccessibilityLevel`, as used [here](https://github.com/Hamsda/EmoTrackerPacks/blob/master/ootrando_overworldmap_hamsda/scripts/logic_entrance.lua#L2)

**Comment by deoxis9001** [comment]
hi
I use emotracker and would like to transfer my pack to poptracker.
But for that to happen AccessibilityLevel.SequenceBreak => would have to work.
AccessibilityLevel.SequenceBreak => out of logic
AccessibilityLevel.Inspect is only for capture item #12 
https://github.com/deoxis9001/tmcrando_maptracker_deoxis/blob/Stable/scripts/locations/Castle.lua

**Comment by black-sliver** [comment]
Related discussion on Discord starting here: <https://discord.com/channels/937157230963339364/937163459534127174/1345850590365749331>

Currently the Lua functions go through ProviderCountForCode, same as all other codes, for simplicity and caching. Provider Count can not be anything other than a number (0 or 1), i.e. not a tuple of integer + AccessibilityLevel.
The issue was not references/mentioned again since I saw a pack doing that, so I did not give it any attention.

There is a special syntax `^$` to call a Lua function that directly returns the AccessibilityLevel for a location/section, and nothing else, however that is not back compatible to Emo.
The alternative that currently works in both, I believe, would be adding 2 Lua calls to the location/section rules:
`"[$rule_in_logic], $rule_with_sequence_breaks"`.

Let me know if you think we should absolutely implement the Emo interface. It's not off the table because I think compatibility to existing packs is important, even though I do not agree with the design decision (which is why we have `^$`, which makes a lot more sense in my opinion).
